modules/consensus/consistency.go

```diff
@@ -2,281 +2,331 @@ package consensus
+	var claimSiacoins types.Currency
+	err = tx.Bucket(SiafundOutputs).ForEach(func(_, sfoBytes []byte) error {
+		var sfo types.SiafundOutput
+		err := encoding.Unmarshal(sfoBytes, &sfo)
+		if build.DEBUG && err != nil {
+			panic(err)
+		}
 
-	// Add all the blocks in the current path TODO: along with their diffs.
-	for i := 0; i < int(cs.db.pathHeight()); i++ {
-		tree.PushObject(cs.db.getPath(types.BlockHeight(i)))
+		coinsPerFund := getSiafundPool(tx).Sub(sfo.ClaimStart)
+		claimCoins := coinsPerFund.Mul(sfo.Value).Div(types.SiafundCount)
+		claimSiacoins = claimSiacoins.Add(claimCoins)
+		return nil
+	})
+	if err != nil {
+		panic(err)
 	}
 
-	// Add all of the siacoin outputs, sorted by id.
-	var openSiacoinOutputs crypto.HashSlice
-	cs.db.forEachSiacoinOutputs(func(scoid types.SiacoinOutputID, sco types.SiacoinOutput) {
-		openSiacoinOutputs = append(openSiacoinOutputs, crypto.Hash(scoid))
-	})
-	sort.Sort(openSiacoinOutputs)
-	for _, id := range openSiacoinOutputs {
-		sco := cs.db.getSiacoinOutputs(types.SiacoinOutputID(id))
-		tree.PushObject(id)
-		tree.PushObject(sco)
+	// Count how many coins should exist
+	deflationBlocks := types.BlockHeight(types.InitialCoinbase - types.MinimumCoinbase)
+	expectedSiacoins := types.CalculateCoinbase(0).Add(types.CalculateCoinbase(blockHeight(tx))).Div(types.NewCurrency64(2))
+	if blockHeight(tx) < deflationBlocks {
+		expectedSiacoins = expectedSiacoins.Mul(types.NewCurrency64(uint64(blockHeight(tx) + 1)))
+	} else {
+		expectedSiacoins = expectedSiacoins.Mul(types.NewCurrency64(uint64(deflationBlocks + 1)))
+		trailingSiacoins := types.NewCurrency64(uint64(blockHeight(tx) - deflationBlocks)).Mul(types.CalculateCoinbase(blockHeight(tx)))
+		expectedSiacoins = expectedSiacoins.Add(trailingSiacoins)
 	}
 
-	// Add all of the file contracts, sorted by id.
-	var openFileContracts crypto.HashSlice
-	cs.db.forEachFileContracts(func(fcid types.FileContractID, fc types.FileContract) {
-		openFileContracts = append(openFileContracts, crypto.Hash(fcid))
-	})
-	sort.Sort(openFileContracts)
-	for _, id := range openFileContracts {
-		// Sanity Check - file contract should exist.
-		fc := cs.db.getFileContracts(types.FileContractID(id))
-		tree.PushObject(id)
-		tree.PushObject(fc)
+	totalSiacoins := dscoSiacoins.Add(scoSiacoins).Add(fcSiacoins).Add(claimSiacoins)
+	if totalSiacoins.Cmp(expectedSiacoins) != 0 {
+		fmt.Println("Wrong number of siacoins... diagnostics:")
+		fmt.Println("Dsco:", dscoSiacoins)
+		fmt.Println("Sco:", scoSiacoins)
+		fmt.Println("Fc:", fcSiacoins)
+		fmt.Println("Claim:", claimSiacoins)
+		if totalSiacoins.Cmp(expectedSiacoins) < 0 {
+			fmt.Println("total:", totalSiacoins)
+			fmt.Println("expected:", expectedSiacoins)
+			fmt.Println("expected is bigger:", expectedSiacoins.Sub(totalSiacoins))
+		} else {
+			fmt.Println("total:", totalSiacoins)
+			fmt.Println("expected:", expectedSiacoins)
+			fmt.Println("total is bigger:", totalSiacoins.Sub(expectedSiacoins))
+		}
```

Comment:
instead of printing directly, stick these in a string and call `panic` on the string